simple-facebook-instant-articles.php

```diff
@@ -192,6 +205,103 @@ public function gallery( $atts, $content = '' ) {
 		<?php return ob_get_clean();
 	}
 
+	/**
+	 * Hook into the_content filter, so the post content
+	 * can be rendered into FB IA format.
+	 */
+	public function render_post_content() {
+		add_filter( 'the_content', array( $this, 'fb_formatted_post_content' ) );
+	}
+
+	/**
+	 * Setup dom and xpath objects for formatting post content.
+	 * Introduces `simple_fb_formatted_post_content` filter, so that post content
+	 * can be formatted as necessary and dom/xpath objects re-used.
+	 *
+	 * @param $post_content Post content that needs to be formatted into FB IA format.
+	 *
+	 * @return string|void  Post content in FB IA format if dom is generated for post content,
+	 *                      Otherwise, nothing.
+	 */
+	public function fb_formatted_post_content( $post_content ) {
+
+		$dom = new \DOMDocument();
+
+		// Parse post content to generate DOM document.
+		// Use loadHTML as it doesn't need to be well-formed to load.
+		@$dom->loadHTML( '<html><body>' . $post_content . '</body></html>' );
+
+		// Stop - if dom isn't generated.
+		if ( ! $dom ) {
+			return;
+		}
+
+		$xpath = new \DOMXPath( $dom );
+
+		// Allow to render post content via action.
+		do_action_ref_array( 'simple_fb_formatted_post_content', array( &$dom, &$xpath ) );
+
+		// Get the FB formatted post content HTML.
+		$body_node = $dom->getElementsByTagName( 'body' )->item( 0 );
+		return $this->get_html_for_node( $body_node );
+	}
+
+	/**
+	 * Renders pull quotes into FB AI format.
+	 * Ref: https://developers.facebook.com/docs/instant-articles/reference/pullquote
+	 *
+	 * @param DOMDocument $dom   Dom object generated for post content.
+	 * @param DOMXPath    $xpath Xpath object generated for post content.
+	 */
+	public function render_pull_quotes( \DOMDocument &$dom, \DOMXPath &$xpath ) {
+
+		// Pull quotes - with <cite> element.
+		foreach ( $xpath->query( '//blockquote[descendant::cite]' ) as $node ) {
```

Comment:
I guess we need to handle blockquotes that don't have a `cite` element nested too. 

I wonder if we should loop through all blockquotes, and then check whether they handle `cite` and handle as neccessary? I suspect this will be more performant than doing a second query/loop for all blockquotes that DONT have cite?